src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -1533,6 +1533,159 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_ExpressionTreeContainsAssignment, "y = y").WithLocation(9, 45));
         }
 
+        [Fact, WorkItem(30776, "https://github.com/dotnet/roslyn/issues/30776")]
+        public void RefStructExpressionTree()
+        {
+            var text = @"
+using System;
+using System.Linq.Expressions;
+public class Class1
+{
+    public void Method1()
+    {
+        Method((Class1 c) => c.Method2(default(Struct1)));
+    }
+
+    public void Method2(Struct1 s1) { }
+
+    public static void Method<T>(Expression<Action<T>> expression) { }
+}
+
+public ref struct Struct1 { }
+";
+            var compilation = CreateCompilationWithMscorlib40AndSystemCore(text).VerifyDiagnostics(
+                // (8,40): error CS7053: An expression tree may not contain 'ref struct'
+                //         Method((Class1 c) => c.Method2(default(Struct1)));
```

Comment:
Not a fan of this error message. I'd add a new one that says something like, "expression trees may not contain values of `ref struct` type." It's not too hard to add a resource and it would be much clearer.

Reply:
Does this mean adding a new error code? Anything I should know about how creating new ErrorCode values works?

Reply:
Yup, it basically means you add a new error to the bottom of ErrorCode.cs and add a corresponding resource with the same name as the enum value to CSharpResources.resx (use an XML or simple text editor) at the bottom.